LogBlockIndexWriterTest is flaky

```
14:52:27 [1 - Java Tests] [ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.412 s <<< FAILURE! - in io.zeebe.logstreams.log.LogBlockIndexWriterTest
14:52:27 [1 - Java Tests] [ERROR] shouldAppendMultipleBlocks(io.zeebe.logstreams.log.LogBlockIndexWriterTest)  Time elapsed: 0.484 s  <<< FAILURE!
14:52:27 [1 - Java Tests] org.junit.ComparisonFailure: expected:<4294967[808]L> but was:<4294967[680]L>
14:52:27 [1 - Java Tests] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
14:52:27 [1 - Java Tests] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
14:52:27 [1 - Java Tests] 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
14:52:27 [1 - Java Tests] 	at io.zeebe.logstreams.log.LogBlockIndexWriterTest.assertCorrectBlockAndEvent(LogBlockIndexWriterTest.java:291)
14:52:27 [1 - Java Tests] 	at io.zeebe.logstreams.log.LogBlockIndexWriterTest.shouldAppendMultipleBlocks(LogBlockIndexWriterTest.java:125)
14:52:27 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
14:52:27 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
14:52:27 [1 - Java Tests] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
14:52:27 [1 - Java Tests] 	at java.lang.reflect.Method.invoke(Method.java:498)
14:52:27 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
14:52:27 [1 - Java Tests] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
14:52:27 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
14:52:27 [1 - Java Tests] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
14:52:27 [1 - Java Tests] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
14:52:27 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
14:52:27 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
14:52:27 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
14:52:27 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
14:52:27 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
14:52:27 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
14:52:27 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:128)
14:52:27 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:27)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
14:52:27 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
14:52:27 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
14:52:27 [1 - Java Tests] 
```

```
5:08:08 [1 - Java Tests] [ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.075 s <<< FAILURE! - in io.zeebe.logstreams.log.LogBlockIndexWriterTest
15:08:08 [1 - Java Tests] [ERROR] shouldAppendBlockAfterRecover(io.zeebe.logstreams.log.LogBlockIndexWriterTest)  Time elapsed: 0.485 s  <<< FAILURE!
15:08:08 [1 - Java Tests] org.junit.ComparisonFailure: expected:<[8589934592]L> but was:<[4294967296]L>
15:08:08 [1 - Java Tests] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
15:08:08 [1 - Java Tests] 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
15:08:08 [1 - Java Tests] 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
15:08:08 [1 - Java Tests] 	at io.zeebe.logstreams.log.LogBlockIndexWriterTest.assertCorrectBlockAndEvent(LogBlockIndexWriterTest.java:291)
15:08:08 [1 - Java Tests] 	at io.zeebe.logstreams.log.LogBlockIndexWriterTest.shouldAppendBlockAfterRecover(LogBlockIndexWriterTest.java:227)
15:08:08 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:08:08 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:08:08 [1 - Java Tests] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:08:08 [1 - Java Tests] 	at java.lang.reflect.Method.invoke(Method.java:498)
15:08:08 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
15:08:08 [1 - Java Tests] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
15:08:08 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
15:08:08 [1 - Java Tests] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
15:08:08 [1 - Java Tests] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
15:08:08 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:08:08 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:08:08 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:08:08 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
15:08:08 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
15:08:08 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
15:08:08 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:128)
15:08:08 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:27)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
15:08:08 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
15:08:08 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

--- a/logstreams/src/main/java/io/zeebe/logstreams/impl/log/index/LogBlockIndex.java
+++ b/logstreams/src/main/java/io/zeebe/logstreams/impl/log/index/LogBlockIndex.java
@@ -145,11 +145,11 @@ public class LogBlockIndex {
       throw new IllegalArgumentException(errorMessage);
     }
 
-    lastVirtualPosition = blockPosition;
     final DbLong dbBlockPosition = indexContext.writeKeyInstance(blockPosition);
     final DbLong dbBlockAddress = indexContext.writeValueInstance(blockAddress);
 
     indexColumnFamily.put(indexContext.getDbContext(), dbBlockPosition, dbBlockAddress);
+    lastVirtualPosition = blockPosition;
   }
 
   /**

